flowmachine/flowmachine/core/cache.py

```diff
@@ -238,14 +243,14 @@ def reset_cache(connection: "Connection", redis: StrictRedis) -> None:
     # they point to a cache table and hence are a duplicate of a Query entry which
     # will also point to that table.
 
-    qry = f"SELECT tablename FROM cache.cached WHERE NOT class='Table'"
-    tables = connection.fetch(qry)
     with connection.engine.begin() as trans:
+        qry = f"SELECT tablename FROM cache.cached WHERE NOT class='Table'"
+        tables = connection.fetch(qry)
         trans.execute("SELECT setval('cache.cache_touches', 1)")
         for table in tables:
             trans.execute(f"DROP TABLE IF EXISTS cache.{table[0]} CASCADE")
-        trans.execute("TRUNCATE cache.cached CASCADE")
         trans.execute("TRUNCATE cache.dependencies CASCADE")
+        trans.execute(f"DELETE FROM cache.cached WHERE NOT class='Table'")
```

Comment:
This is more of a meta-comment (and certainly outside the scope of this PR), but somehow it doesn't quite feel right to me that we need to sprinkle functions with special cases for `Table`. It introduces a dependency of these functions on the inner workings of the query hierarchy - in particular, how queries store themselves - which it feels would be nice to avoid. (Also, what if other query classes in the future introduce other special storing behaviour?)

I'm just leaving this here as a comment for further consideration. In case we can find a way of moving this responsibility into the query objects themselves to avoid these kinds of special handlings elsewhere in the code this would probably be useful.